### PR TITLE
Improve error feedback for when playing before the session started

### PR DESCRIPTION
**Pull request by henriquegemignani**

### CODE REVIEW

**Review by ThanatosGit — CHANGES_REQUESTED** [review]
I tried to follow the flow.
Is that still possible?
https://github.com/randovania/randovania/blob/b03741db9285abf026dc1b611b5c2dd7ea0ccf88/randovania/gui/multiworld_client.py#L167C55-L167C55

It looks to me that `_create_new_sync_request` is removing these uuids and the only way to get it back into a request is by `on_session_meta_update`.

**Comment by henriquegemignani** [comment]
> I tried to follow the flow. Is that still possible? https://github.com/randovania/randovania/blob/b03741db9285abf026dc1b611b5c2dd7ea0ccf88/randovania/gui/multiworld_client.py#L167C55-L167C55
> 
> It looks to me that `_create_new_sync_request` is removing these uuids and the only way to get it back into a request is by `on_session_meta_update`.

Good catch.

The code was from an old version where `_world_sync_errors` was just for tracking the sessions that got something that isn't `WorldDoesNotExistError`/`WorldNotAssociatedError` so they could be displayed, but then I realised it could be merged better
Requests that get a ServerError should still be retried (though I need to add some sort of exponential backoff...)

File: test/gui/test_multiworld_client.py [hunk 1]
```diff
@@ -278,37 +278,50 @@ async def test_server_sync(client, mocker: MockerFixture):
             session_name="The Session",
         )
     )
-    assert client._blacklisted_worlds == {
+    assert client._world_sync_errors == {
         uid_3: error.WorldDoesNotExistError(),
     }
 
 
 async def test_on_session_meta_update_not_logged_in(client: MultiworldClient):
```
[the remark this answers is not in the document]
Comment (henriquegemignani) [reply test/gui/test_multiworld_client.py:287]: If you see the line `client.network_client.perform_world_sync` above, in the second call for that we're mocking a response that `uid_3` fails with `WorldDoesNotExistError`.

Then we assert that end of the process this causes `uid_3` to be marked with that error in `_world_sync_errors `.

[the remark this answers is not in the document]
Comment (ThanatosGit) [reply test/gui/test_multiworld_client.py:287]: Uh, I think you are referencing the wrong test. Comments are little bit weird here. The comment belongs to L286 and following. To clarify: I mean the `test_on_session_meta_update_not_logged_in`

[the remark this answers is not in the document]
Comment (henriquegemignani) [reply test/gui/test_multiworld_client.py:287]: Ah yes. You made the comment in the test title.

I kinda just threw stuff around in the existing tests without thinking much, but this is still a test where `network_client.current_user` is None. For me it's testing:
- function doesn't blow up in that case
- it still clears `WorldDoesNotExistError` errors even if you're not logged in, as long as you get session events.

of course, you don't get session events if you don't have an open session and that requires you to be logged in, but I'm making sure MultiworldClient would work if that changes

**Review by ThanatosGit — APPROVED** [review]

**Pull request merged**